src/Rules/Classes/ClassConstantRule.php

```diff
@@ -60,6 +61,24 @@ public function processNode(Node $node, Scope $scope): array
 				$name = $constantString->getValue();
 				$constantNameScopes[$name] = $scope->filterByTruthyValue(new Identical($node->name, new String_($name)));
 			}
+
+			if ($this->checkNonStringableDynamicAccess) {
+				$accepts = $this->ruleLevelHelper->accepts(new StringType(), $nameType, true);
+				$typeResult = $this->ruleLevelHelper->findTypeToCheck(
+					$scope,
+					$node->name,
+					'',
+					static fn (Type $type) => $type->toString()->isString()->yes()
+				);
```

Comment:
After this call there should be:

```php
		$type = $typeResult->getType();
		if ($type instanceof ErrorType) {
			return [];
		}
```

Some rules return "unknown class errors" but that's irrelevant here.

Reply:
Dynamic fetching of constants doesn't do implicit casting, so it seems appropriate to simply check `isString()->yes()` without `toString()`.

I was wrong in the early stages of this PR.

Reply:
I'd rename the variable to `$nameTypeResult`.